### PR TITLE
Id 37 reset password openapi

### PR DESCRIPTION
**Pull request by bdsl**
Added open api annotations to the three controllers, and also copied the code from mailer to output a copy of the swagger UI as a CircleCI artifeact so we can view the API docs.

### CODE REVIEW

**Comment by bdsl** [comment]
Updated docs can be viewed at https://output.circle-artifacts.com/output/job/80754e78-f2b0-4289-afc2-eba9c935eed1/artifacts/0/docs/api-docs.html#/default

File: src/Application/Actions/CreatePasswordResetToken.php [hunk 1]
```diff
@@ -8,14 +8,33 @@
...
+ *     path="/v1/password-reset-token",
+ *      @OA\RequestBody(
+ *         description="",
+ *         required=true,
+ *         @OA\JsonContent(
+ *              @OA\Property(property="email_address", type="string", example="fred@example.com"),
+ *        )
+ *      ),
+ *     @OA\Response(
+ *         response=200,
```
Comment (NoelLH) [inline src/Application/Actions/CreatePasswordResetToken.php:30]: As above, I think where the action specifically has logic branches which 400, it's good to cover that in the doc

**Review by NoelLH — COMMENTED** [review]
Looking good! Just a couple of places I think we should be expanding on potential errors

File: src/Application/Actions/CreatePasswordResetToken.php [hunk 2]
```diff
@@ -8,14 +8,38 @@
...
+ *         description="Token created and emailed to user, if they exist, which the server does not confirm or deny via http",
+ *         @OA\JsonContent(),
+ *        ),
+ *     @OA\Response(
+ *         response=400,
+ *         description="Email address is invalid, e.g. doesn't create @ sign. The absecne of this error doesn't mean the eamil is registered with us or exists, just that the format looks OK.. ",
```
Comment (NoelLH) [inline src/Application/Actions/CreatePasswordResetToken.php:36]:
```suggestion
 *         description="Email address is invalid, e.g. doesn't create @ sign. The absence of this error doesn't mean the eamil is registered with us or exists, just that the format looks OK.. ",
```

**Comment by bdsl** [comment]
@NoelLH Thanks. I've added docs for the 400 response. Just need to work out why the checkout step is failing now.

**Review by NoelLH — APPROVED** [review]

**Comment by bdsl** [comment]
Updated docs: https://output.circle-artifacts.com/output/job/a45aeba8-68a5-4490-b761-cfbe0e8c2689/artifacts/0/docs/api-docs.html

**Pull request merged**